modules/nlu/src/backend/api.ts

```diff
@@ -156,6 +156,24 @@ export default async (bp: typeof sdk, state: NLUState) => {
     }
   })
 
+  router.post('/intents/:intentName', async (req, res) => {
+    const botEngine = state.nluByBot[req.params.botId].engine1 as ScopedEngine
+    const targetName = req.params.intentName
+    const intent = await botEngine.storage.getIntent(targetName)
+    intent.name = req.body.name
+    await botEngine.storage.updateIntent(targetName, intent)
+    scheduleSyncNLU(req.params.botId)
+
+    res.sendStatus(200)
+  })
+
+  router.post('/intents/:intentName/duplicate', async (req, res) => {
```

Comment:
I would rather move the duplicate piece of logic in the front end and use the post to /intents instead of creating this in the storage (which by the way, will be removed soon)

Reply:
You are right, it can be done from the front end.

Oh, then I will use `/intents` before it goes away :)